src/bin/main.rs

```diff
@@ -4,19 +4,23 @@ extern crate log;
 use actix_web::{App, HttpServer};
 
 use hilow::router::router;
+use hilow::state::init_app_state;
 
 #[actix_web::main]
-async fn main() -> std::io::Result<()> {
+async fn main() -> anyhow::Result<()> {
     if cfg!(debug_assertions) {
         dotenv::from_filename(".env.development")
             .expect("Unable to find \".env.development\" in the CWD");
     }
 
     env_logger::init();
 
+    let app_state = init_app_state().await?;
+
     info!("Serving on http://0.0.0.0:7878");
-    HttpServer::new(|| App::new().configure(router))
+    HttpServer::new(move || App::new().data(app_state.clone()).configure(router))
```

Comment:
This `clone` here don't seems right, we want `App` to be the owner of `app_state` now theres no need to clone it then

Reply:
I think that the clone is needed here, each worker of the thread should own their own reference to the Arc<State> struct.

api_1       | 21 |     HttpServer::new(move || App::new().data(app_state).configure(router))
api_1       |    |                                             ^^^^^^^^^ move occurs because `app_state` has type `Arc<State>`, which does not implement the `Copy` trait

Reply:
This is happening because you are binding `app_state` to `main` lifetime. What happens if you declare `app_state` inside of `data` call?

```suggestion
    HttpServer::new(move || App::new().data(init_app_state().await?).configure(router))
```

Note: I didn't tried this locally